lambdas/service/app.py

```diff
@@ -655,204 +661,237 @@ def repository_search(entity_type: str, item_id: str):
+                for param in ['search_before', 'search_before_uid', 'search_after', 'search_after_uid']
+            ]
+        ],
+        'responses': {
+            '200': {
+                'description': format_description(f'''
+                    Paginated list of {entity_type}s that meet the search
+                    criteria ("hits"). The structure of these hits is documented
+                    under the [corresponding endpoint for a specific entity]({id_spec_link}).
+
+                    The `pagination` section describes the total number of hits
+                    and total number of pages, as well as user-supplied search
+                    parameters for page size and sorting behavior. It also
+                    provides links for navigating forwards and backwards between
+                    pages of results.
+
+                    The `termFacets` section tabulates the occurrence of unique
+                    values within nested fields of the `hits` section. Not every
+                    nested field is tabulated, but the set of tabulated fields
+                    is consistent between entity types.
+                '''),
+                **schema.wrap_json_schema_content(page_spec)
+            }
+        }
+    }
+
+
+def repository_id_spec(entity_type: str):
+    search_spec_link = f'#operations-Repository-get_repository_{entity_type}s_'
+    return {
+        'summary': f'Detailed information on a particular {entity_type} entity.',
+        'tags': ['Repository'],
+        'parameters': [
+            params.path(f'{entity_type}_id', str, description=f'The UUID of the desired {entity_type}')
+        ],
+        'responses': {
+            '200': {
+                'description': format_description(f'''
+                    This response describes a single {entity_type} entity. To
+                    search the repository for multiple entities, see the
+                    [corresponding search end endpoint]{search_spec_link}.
+
+                    The properties that are common to all entity types are
+                    listed in the schema below; however, additional properties
+                    may be present for certain entity types. With the exception
+                    of the {entity_type}'s unique identifier, all properties are
+                    arrays, even in cases where only one value is present.
+
+                    The inner structures of the objects within these arrays are
+                    not consistent between entity types, e.g. projects and
+                    files, and in a very small number of cases, not consistent
+                    between entities of the same type, e.g. between two
+                    different files.
+                '''),
+                **schema.wrap_json_schema_content(hit_spec)
+            }
+        }
+    }
+
+
+repository_head_spec = {
+    'summary': 'Used by the /health endpoint for monitoring application status.',
```

Comment:
That's an implementation detail.

```suggestion
    'summary': 'Perform a query without returning its result.',
```

The description should say sth like "Can be used to check the validity of query parameters or to test whether the {entity_type} index is operational."